tests/test_util.py

```diff
@@ -78,3 +83,42 @@ def test_copy_file_bulk(self) -> None:
                             call("/src/path/foo/file2", "/target/path/foo/file2"),
                         ]
                     )
+
+
+def _response_generator(target_value: int) -> Callable[[Request], Response]:
+    """Return a generator with closure so a value can be tracked across multiple invocations."""
+    value = 0
+
+    def generator(request: Request) -> Response:  # pylint: disable=unused-argument
+        """Add the next value as a header and adjust the status code based on the value."""
+        nonlocal value, target_value
+        value += 1
+        response = Response()
+        response.status_code = 403 if value <= (target_value + 1) else 200
+        response.headers["X-VALUE"] = str(value)
+        return response
+
+    return generator
+
+
+def test_get_http_failure(httpserver: HTTPServer) -> None:
```

Comment:
To make the goal of this test more clear, it's better to split it to two test functions and describe the goal and expected behavior.